app/gui/src/project-view/util/data/observable.ts

```diff
@@ -2,7 +2,18 @@
 
 import type { ObservableV2 } from 'lib0/observable'
 
+/** Possible events which can be emitted by Observable. */
 export type Events<O extends ObservableV2<any>> = O extends ObservableV2<infer E> ? E : never
+/** A list of all possible events of the observable, in fixed (but unspecified) order. */
+export type ListOfEvents<O extends ObservableV2<any>> = TupleFromUnion<keyof Events<O>>
+
+type TupleFromUnion<T, L = LastOf<T>, N = [T] extends [never] ? true : false> =
+  true extends N ? [] : [...TupleFromUnion<Exclude<T, L>>, L]
+type UnionToIntersection<U> =
+  (U extends any ? (k: U) => void : never) extends (k: infer I) => void ? I : never
+type LastOf<T> =
+  UnionToIntersection<T extends any ? (x: T) => void : never> extends (x: infer L) => void ? L
```

Comment:
I admit I don't understand this. Why this `UnionToIntersection` takes a function type, not a union? Any chance of explanation?

Reply:
Nope, not a chance.

After researching it a bit, I think the current implementation is too fragile to use, I will come up with something else.